src/main/java/fr/sncf/osrd/infra/SuccessionTable.java

```diff
@@ -0,0 +1,29 @@
+package fr.sncf.osrd.infra;
+
+import java.util.ArrayList;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+/**
+ * The mutable succesion table for a switch (order of trains on that switch)
+ * There must be a SuccesionTable instance per switch on the network.
+ */
+
+@SuppressFBWarnings({ "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD" })
```

Comment:
This warning suppression should not be needed, and if it were, the warning would be important.